src/Data/IO.php

```diff
@@ -116,11 +115,10 @@ public function sequence(ApplicativeInstance $fa): ApplicativeInstance
         }
 
         // todo: rewrite this in a more concise manner, but for brevity now:
-        $do = function () use ($fa) {
-            $f = $this();
-            $g = $fa();
-            return partial ($f) ($g);
-        };
+        $do = fn ()
+            => null |> $fa
+            |> partial (null |> $this);
```

Comment:
I think this would look better like

```suggestion
        $do = fn () => $fa() |> partial ($this());
```